code/game/objects/structures/stool_bed_chair_nest/stools.dm

```diff
@@ -74,3 +73,15 @@
 		T.Weaken(5)
 		return
 	..()
+
+/obj/item/weapon/stool/dropped(mob/user) //Make sure we aren't leaving un-deconstructible stools.
+	..()
+	if(!user.in_throw_mode) //If this stool isn't getting thrown, turn it back into a structure.
+		origin.forceMove(user.loc)
+		qdel(src)
+
+/obj/item/weapon/stool/throw_impact(atom/hit_atom) //Thrown but uncaught stools will automatically turn into the structure version as soon as the throw is over.
+	..()
+	if(!iscarbon(loc)) //If it wasn't caught by someone, put the structure version down.
+		new origin.type(get_turf(src))

```

Comment:
It's not getting moved correctly because its loc is getting set before the throw processing is entirely finished. If I throw it onto an empty space for example, there is little processing due to there not being any hit object. As soon as I throw the stool at, say, a window, it'll poof into whereverspace as the loc still gets set at about the time as it would if it were thrown at an empty floor tile, but there's still processing on the part of the hit object to deal with.